src/sinks/datadog/logs/config.rs

```diff
@@ -45,33 +45,55 @@ impl SinkBatchSettings for DatadogLogsDefaultBatchSettings {
     const TIMEOUT_SECS: f64 = BATCH_DEFAULT_TIMEOUT_SECS;
 }
 
-#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+/// Configuration for the `datadog_logs` sink.
+#[configurable_component(sink)]
+#[derive(Clone, Debug, Default)]
 #[serde(deny_unknown_fields)]
-pub(crate) struct DatadogLogsConfig {
+pub struct DatadogLogsConfig {
+    /// The endpoint to send logs to.
     pub(crate) endpoint: Option<String>,
-    // Deprecated, replaced by the site option
+
+    /// The Datadog region to send logs to.
+    #[configurable(deprecated)]
```

Comment:
Should we still be noting what has superseded the deprecated option?

Reply:
Hmm, probably.

My intent is to add support to the `deprecated` helper attribute that lets you specify the name of another field on the object that replaces it... but it has some edge cases (like if the field only exists via a `#[serde(flatten)]`-ed field, since ideally we would want to enforce that the field you're saying supersedes it actually exists.

The flipside is that the intent is to also use `#[serde(alias = "...")]` as a signal for deprecated fields, since we should never allow two identical ways to set a field save for backwards compatibility, and a backwards compatible field implies that it is deprecated.

So I'm not sure, we do need a way to support both, because if the field is an entirely different type, you can't just alias it, but it would also be nice to have a consistent way to mark/demonstrate a field being deprecated.

Long story short, I'll update the comment to refer to the field that supersedes it.

Reply:
Yeah, it seemed like `alias` was a "better" way of doing it, assuming it was a name change and not changing the provided contents as well. I can see how it'd be tricky to make work universally.

Not knowing what you need to replace your deprecated option with is definitely a bad time.

Reply:
Fixed the comment in a3fff3e3d.